src/types/Navigation.d.ts

```diff
@@ -22,12 +22,13 @@ type NavBarItem = {
    */
   className: "navMobileItem" | "navMobileItem clickable";
   /**
-   * Defines RBAC for the Navigation Item
+   * Defines PBAC for the Navigation Item via a callback function
    *
    * Guide:
-   * - If the value is an array, the current user must be in one of the roles to see the item
+   * - Provide a function that returns true if the NavBarItem should be shown,
+   *   otherwise it will be hidden
    */
-  roles?: User["role"][];
+  hasPermission?: (user: User) => boolean;
```

Comment:
You don't need to make any changes here, but what are your thoughts on using just an array of required permissions instead of a function? e.g.

```suggestion
  permissions?: [AuthPermissions];
```

Reply:
Agreed, added it to the sub items as well. Although, the NavbarDesktop just hard-codes each of the name dropdown nav items. (Ex. "User Profile", "Uploader CLI Tool", etc.) That would require some refactoring to support the same functionality there. 

The rest were updated in [2af0290](https://github.com/CBIIT/crdc-datahub-ui/pull/567/commits/2af02901e83aff22765243aa674c55fd3c0b26d3).

Reply:
Thanks for updating that. That hardcoding has always been messy (it really should be in the config IMO). Maybe I can address this with CRDCDH-2124.